### ISSUE TITLE
Ablage der Daten in anderen Datenbanken

### ISSUE THREAD

**Issue by mdzio** [mdzio/ccu-historian#36]
* MySQL
* PostgreSQL

**Comment by mbirth** [comment]
Ich würde noch SQLite3 in den Topf werfen.

**Comment by mdzio** [comment]
Das Datenbanksystem SQLite bietet keine Vorteile gegenüber der bisher verwendeten H2Database. Da SQLite keine Netzwerkschnittstelle besitzt, bietet es funktional sogar weniger.

Es existiert nun ein Prototyp mit MySQL-Anbindung. Aber die Stabilität verschlechtert sich stark, wenn es zu kleineren Netzwerkausfällen kommt.

**Comment by mbirth** [comment]
SQLite hat dafür keinerlei Overhead wie z.B. Userverwaltung oder eben Netzwerkschnittstelle und Webserver. Als reiner Datenspeicher ist das prima geeignet. Wenn MySQL-Support drin ist, ist SQLite ja eigentlich auch "nur noch" eine etwas andere JDBC Connection und das SQL für Erstellung der Tabellen muss einmalig geschrieben/angepasst werden. Die Insert- und Select-Operationen nehmen sich ja nichts.

**Comment by mdzio** [comment]
Ja, die Schnittstelle sollte generisch gestaltet werden, sodass eigentlich alle SQL-Datenbanken mit JDBC-Treiber verwendet werden könnten.

**Comment by mdzio** [comment]
Auf absehbare Zeit wird der CCU-Historian funktional nicht mehr erweitert. Er wird weiterhin gepflegt, sodass er mit allen CCU-Zentralen, auch den kommenden auf OpenCCU-Basis, kompatibel bleibt.
[Weitere Informationen sind auf der Startseite zu finden.](https://github.com/mdzio/ccu-historian?tab=readme-ov-file#historie-und-projektstatus)
